### PR TITLE
Vaurcae now receive language processors/psionic receivers through culture/origin

### PR DESCRIPTION
**Pull request by feartheblackout**
Requested by the Vaurca lore team, specifically @connorjg1. This is, hopefully, a more ideal way of giving Vaurcae their needed augments, while also reducing the loadout point cost burden due to the recent removal of free loadout items.

### CODE REVIEW

**Comment by feartheblackout** [comment]
!review

File: html/changelogs/SimpleMaroon-vaurcalanguagerework.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#   experiment
+#   balance
+#   admin
+#   backend
+#   security
+#   refactor
+#################################
+
+# Your name.
+author: ChangeMe
```
Comment (SleepyGemmy) [inline html/changelogs/SimpleMaroon-vaurcalanguagerework.yml:30]: .

File: code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm [hunk 2]
```diff
@@ -149,6 +149,29 @@
...
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_MIKUETZ_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/mikuetz(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
+	if(HAS_TRAIT(H, TRAIT_ORIGIN_HAS_ZINO_IMPLANT))
+		var/obj/item/organ/A = new /obj/item/organ/internal/augment/language/zino(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm:171]: This is a horridly hardcoded way of handling this. You need to generalize this whole behaviour somehow. Offloading it to the origin and generalizing it to work with all augments is a start.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm (hunk 2 above)
Comment (feartheblackout) [reply code/modules/mob/living/carbon/human/species/station/vaurca/vaurca.dm]: Not sure why it's asking for a changelog again, but change addressed.

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/background/origins/_origins.dm [hunk 3]
```diff
@@ -20,3 +21,9 @@
 /singleton/origin_item/proc/on_apply(var/mob/living/carbon/human/H)
 	for(var/trait in origin_traits)
 		ADD_TRAIT(H, trait, CULTURE_TRAIT)
+
+	for(var/augment in origin_augs)
+		var/obj/item/organ/A = new augment(H)
+		var/obj/item/organ/external/affected = H.get_organ(A.parent_organ)
+		A.replaced(H, affected)
```
Comment (NonQueueingMatt) [inline code/modules/background/origins/_origins.dm:28]: You need to check if affected exists here.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by Arrow768 — CHANGES_REQUESTED** [review]
This has been discussed by the maintainers.

The solution implemented here is a rather limited solution as its restricted to just augments.
Given that it is a very niche mechanic in a core system currently used by a single species in a very important system we do not believe that this is a good implementation.

Ideally we would have the option to assign existing loadout items to characters based on their origin.
With that we are not limited to just augments and open up the for this feature to be used by a lot more origins than just vaurca.

A possible implementation could use a "extra loadout items" list to which origins (possibly amongst other parameters) can add items that are then spawned before the other loadout items.
When implementing care should be taken to ensure a customization of the loadout items (possibly by using a assoc list to store the loadout item path as key and the parameters as value)

**Pull request closed without merging**